src/flow/transactions/mod.rs

```diff
@@ -288,7 +319,52 @@ impl DomainState {
                 self.update_next_transaction();
                 Event::Transaction(v)
             }
+            Bundle::Replay(packet) => {
+                self.ts += 1;
+                self.update_next_transaction();
+                println!("[{:?} @ {}] Replay({})", self.domain_index, self.ts, self.ts);
+                Event::Replay(packet)
+            }
+            Bundle::SeedReplay(tag, key, rts) => {
+                self.ts += 1;
+                self.update_next_transaction();
+                println!("[{:?} @ {}] SeedReplay({})", self.domain_index, self.ts, self.ts);
+                Event::SeedReplay(tag, key, rts)
+            }
             Bundle::Empty => Event::None,
```

Comment:
Couldn't you have this `return`, and then move all the
```rust
self.ts += 1;
self.update_next_transaction();
```
into a single one at the bottom?